core/src/main/java/org/nervos/ckb/methods/type/Transaction.java

```diff
@@ -26,4 +33,48 @@ public Transaction(
     this.outputs = cellOutputs;
     this.witnesses = witnesses;
   }
+
+  public Transaction(
+      String version,
+      String hash,
+      List<OutPoint> deps,
+      List<CellInput> cellInputs,
+      List<CellOutput> cellOutputs,
+      List<Witness> witnesses) {
+    this.version = version;
+    this.hash = hash;
+    this.deps = deps;
+    this.inputs = cellInputs;
+    this.outputs = cellOutputs;
+    this.witnesses = witnesses;
+  }
+
+  public Transaction sign(BigInteger privateKey, String txHash) {
```

Comment:
There could be multiple privateKey. Cells from different lock scripts.

Reply:
You are right, but maybe we are not considering now.